server/statistics/hot_peer_cache.go

```diff
@@ -85,24 +85,24 @@ func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo, stats *StoresSta
 	totalBytes := float64(f.getTotalBytes(region))
 	totalKeys := float64(f.getTotalKeys(region))
 
-	bytesPerSecInit := totalBytes / RegionHeartBeatReportInterval
-	keysPerSecInit := totalKeys / RegionHeartBeatReportInterval
+	reportInterval := region.GetInterval()
+	interval := reportInterval.GetEndTimestamp() - reportInterval.GetStartTimestamp()
+	endTime := time.Unix(int64(reportInterval.GetEndTimestamp()), 0)
+
+	bytesPerSec := totalBytes / float64(interval)
+	keysPerSec := totalKeys / float64(interval)
 
 	for storeID := range storeIDs {
-		bytesPerSec := bytesPerSecInit
-		keysPerSec := keysPerSecInit
 		isExpired := f.isRegionExpired(region, storeID)
 		oldItem := f.getOldHotPeerStat(region.GetID(), storeID)
 
 		// This is used for the simulator.
-		if oldItem != nil && Denoising {
-			interval := time.Since(oldItem.LastUpdateTime).Seconds()
-			// ignore if report too fast
-			if interval < hotRegionReportMinInterval && !isExpired {
+		if oldItem != nil && !isExpired {
+			// ignore if report too fast or an old report
+			isOldReport := endTime.After(oldItem.LastUpdateTime)
```

Comment:
I'm a little bit confused about the meaning of `isOldReport`, does it mean the report time is before the `LastUpdateTime`?

Reply:
Yes, my mistake. Fixed.

Reply:
BTW, do we need to add a test to avoid similar problem